sound/soc/sof/ipc4-topology.c

```diff
@@ -1031,7 +1031,8 @@ static int sof_ipc4_update_hw_params(struct snd_sof_dev *sdev, struct snd_pcm_hw
 static int sof_ipc4_init_output_audio_fmt(struct snd_sof_dev *sdev,
 					  struct sof_ipc4_base_module_cfg *base_config,
 					  struct sof_ipc4_available_audio_format *available_fmt,
-					  int input_audio_format_index)
+					  u32 out_ref_rate, u32 out_ref_channels,
+					  u32 out_ref_valid_bits)
```

Comment:
The commit message is not for the faint-hearted.

````
The only special case is the copier module. In the case of
a module copier, the reference params is based on the selected input
params. 
````

I am not getting the nuance with a regular processing module. Unless 'input audio format' and 'input params' mean different things?

````
In the case of host copier, the reference params should be
based on the input audio format for playback and the FE hw_params for
capture.
````

This one seems fine

````
In the case DAI copier, the reference params should be based on
the input audio format for capture and the FE hw_params for playback.
````

But this one isn't. Why would the DAI parameters depend on the FE hw_params, given that there's likely all kinds of conversions upstream (gain, mixout, what have you). The FE hw_params should only impact the host copier.

Unless you mean BE be_hwparams and this is a typo?